### ISSUE TITLE
OCA\DocumentServer\Document\DocumentConversionException: Couldn't create temp folder 

### ISSUE THREAD

**Issue by autoize** [nextcloud/documentserver_community#21]
This is with the latest version of the Community Document Server and OnlyOffice apps.
Creating or opening any office document hangs on a "document loading" screen with the spinning animation. Have tried uninstalling and reinstalling the apps, and also disabling the "open in new tab" option in the OnlyOffice settings page.

App versions:

- NextCloud 18.0.0.10
- Community Document Server version 0.1.2

This is outputted to the NextCloud log, whenever the problem occurs.

```
OCA\DocumentServer\Document\DocumentConversionException: Couldn't create temp folder

    /var/www/nextcloud/apps/documentserver_community/lib/DocumentConverter.php - line 123:

    OCA\DocumentServer\Document\ConverterBinary->run("/tmp/oc_tmp_7DCyIf-.xml")

    /var/www/nextcloud/apps/documentserver_community/lib/DocumentConverter.php - line 111:

    OCA\DocumentServer\DocumentConverter->runCommand(OCA\Document ... {})

    /var/www/nextcloud/apps/documentserver_community/lib/DocumentConverter.php - line 47:

    OCA\DocumentServer\DocumentConverter->convertFiles("/tmp/oc_tmp_VFezij-.docx", "/var/www/ne ... n")

    /var/www/nextcloud/apps/documentserver_community/lib/Document/DocumentStore.php - line 108:

    OCA\DocumentServer\DocumentConverter->getEditorBinary(null, "docx", "/var/www/ne ... 3")

    /var/www/nextcloud/apps/documentserver_community/lib/XHRCommand/AuthCommand.php - line 141:

    OCA\DocumentServer\Document\DocumentStore->getDocumentForEditor(2100169943, OC\Files\Node\File {}, "docx")

    /var/www/nextcloud/apps/documentserver_community/lib/XHRCommand/CommandDispatcher.php - line 45:

    OCA\DocumentServer\XHRCommand\AuthCommand->handle({ type: "aut ... }}, OCA\Document ... {}, OCA\Document ... {}, OCA\Document ... {}, OCA\Document ... {})

    /var/www/nextcloud/apps/documentserver_community/lib/Channel/Channel.php - line 106:

    OCA\DocumentServer\XHRCommand\CommandDispatcher->handle({ type: "aut ... }}, OCA\Document ... {}, OCA\Document ... {}, OCA\Document ... {})

    /var/www/nextcloud/apps/documentserver_community/lib/Controller/SessionController.php - line 114:

    OCA\DocumentServer\Channel\Channel->handleCommand({ type: "aut ... }})

    /var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php - line 170:

    OCA\DocumentServer\Controller\SessionController->xhrSend(null, "2100169943", "200", "w2p4f1p2")

    /var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php - line 99:

    OC\AppFramework\Http\Dispatcher->executeController(OCA\Document ... {}, "xhrSend")

    /var/www/nextcloud/lib/private/AppFramework/App.php - line 125:

    OC\AppFramework\Http\Dispatcher->dispatch(OCA\Document ... {}, "xhrSend")

    /var/www/nextcloud/lib/private/AppFramework/Routing/RouteActionHandler.php - line 47:

    OC\AppFramework\App::main("OCA\\Docume ... r", "xhrSend", OC\AppFramew ... {}, { documentId ... "})

    <<closure>>

    OC\AppFramework\Routing\RouteActionHandler->__invoke({ documentId ... "})

    /var/www/nextcloud/lib/private/Route/Router.php - line 299:

    call_user_func(OC\AppFramew ... {}, { documentId ... "})

    /var/www/nextcloud/lib/base.php - line 1008:

    OC\Route\Router->match("/apps/docum ... d")

    /var/www/nextcloud/index.php - line 38:

    OC::handleRequest()
```

**Comment by kesselb** [comment]
That seems to be a error from `x2t`. A tool to convert the document.

```
Index: lib/Document/ConverterBinary.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- lib/Document/ConverterBinary.php	(revision 6bd7769a88ae7825c9d920de303c81c925e5d1ea)
+++ lib/Document/ConverterBinary.php	(date 1579636218651)
@@ -44,7 +44,11 @@
 		];
 
 		$pipes = [];
-		$process = proc_open('./x2t ' . escapeshellarg($param), $descriptorSpec, $pipes, self::BINARY_DIRECTORY, []);
+
+		$cmd = './x2t ' . escapeshellarg($param);
+		$this->logger->debug('Run: "' . $cmd . '" to convert document');
+
+		$process = proc_open($cmd, $descriptorSpec, $pipes, self::BINARY_DIRECTORY, []);
 
 		fclose($pipes[0]);
 		$output = stream_get_contents($pipes[1]);
```

I would try the above patch, set loglevel to debug and create or open a document again. The logs should tell you the actual command. Try to run that from cli then and see what happens.

**Comment by autoize** [comment]
Thanks, but applying the patch didn't make a difference.
The log output in "debug" mode was the same as the trace originally posted.
Perhaps can increase the logging level further in config.php, and view the logs directly on the server, rather than in the Log Viewer to see if any further clues.

**Comment by icewind1991** [comment]
should be fixes with 0.1.3

**Comment by autoize** [comment]
Can load the document now in 0.1.3 but the fonts are incorrect.
The Normal style should be Arial 11pt, but it is being displayed as Comic Sans.
Will open a separate case for this.
